src/pages/index.js

```diff
@@ -53,67 +54,206 @@ newCardPopup.setEventListeners();
+      {
+        items: result.reverse(),
+        renderer: createCard,
+      },
+      "#cards__list"
+    );
+    section.renderItems(result);
+  })
+  .catch((err) => {
+    console.error(err); // log the error to the console
+  });
 
-// FUNCTIONS
+// fetch and render user info
+api
+  .getUserInfo()
+  .then((result) => {
+    // update UI to reflect the user info from the db
+    // console.log(result);
+    userInfo.setUserInfo(result.name, result.about);
+    userInfo.setAvatar(result.avatar);
+  })
+  .catch((err) => {
+    console.error(err); // log the error to the console
+  });
+
+//////////////////////////////////////////////////////////
+
+// =========  FUNCTIONS  ========= //
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLikeClick,
+    handleUnlikeClick
+  );
   return card.getView();
 }
 
-//render card
 function renderCard(cardData) {
   const cardView = createCard(cardData);
+  const section = new Section({}, "#cards__list");
   section.addItem(cardView);
 }
 
-// declare a function to handle image click
 function handleImageClick({ name, link }) {
+  // open the popup with the image
   popupWithImage.open({ link, name });
 }
 
-// EVENT HANDLERS
+// =========  EVENT HANDLERS  ========= //
 
 // Handles edit form submit
-function handleProfileEditSubmit() {
-  editFormValidator.disableSubmitButton();
-  editProfilePopup.close();
+function handleProfileEditSubmit({ title, description }) {
+  api
+    .editProfileInfo({
+      name: title,
+      about: description,
+    })
+    .then((data) => {
+      userInfo.setUserInfo(data.name, data.about);
+      editFormValidator.disableSubmitButton();
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 
+// handles new card submission
 function handleNewCardSubmit(name, link) {
-  renderCard({ name, link }, placesWrap);
-  cardFormValidator.disableSubmitButton();
-  newCardPopup.close();
+  api
+    .addNewCard({
+      name: name,
+      link: link,
+    })
+    .then((data) => {
+      renderCard(data);
+      cardFormValidator.disableSubmitButton();
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+// handles deleting card
+function handleDeleteCard(card) {
+  // open the delete card popup
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitCallback(() => {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.deleteCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+// function to handle like button click
+function handleLikeClick(card) {
+  // const cardId = card.getId();
+  api
+    .likeCard(card._id)
+    .then((data) => {
+      card._handleLikeIcon();
+      card._isLiked = true;
```

Comment:
`_handleLikeIcon`
`_isLiked`

Please, delete the **low dash** (`underscore`) from methods if you want to use them `outside` a class because the `low dash` shows a **private** method which is used only inside the class.